Add Mini Apps series (#3938)
GoogleChrome/web.dev: src/site/content/en/mini-apps/mini-app-about/index.md

---
layout: post
title: What are mini apps?
authors:
  - thomassteiner
date: 2021-02-25
# updated: 2021-02-25
description: |
  This chapter introduces the concept of mini apps and provides examples of their look and feel.
tags:
  - mini-apps
---

## Building blocks and compatibility

Mini apps are small (commonly less than 4MB) apps that require a
[super app](/mini-app-super-apps/#for-mini-apps-you-need-super-apps)
to run. What they have in
common, independent of the super app, is that they are built with ("dialects" of) the web
technologies HTML, CSS, and JavaScript. The runtime of a mini app is a
[WebView](https://research.google/pubs/pub46739/) in the super app, not the underlying operating

> system, which means mini apps are cross platform. What this means is that the same mini app can run
> in the same super app, no matter if the super app runs on Android, iOS, or another OS. However, not
> all mini apps can run in all super apps, more on this [later](/mini-app-standardization/).

Reviewer: ```suggestion
system, which makes mini apps cross platform. The same mini app can run
in the same super app, no matter if the super app runs on Android, iOS, or another OS. However, not
all mini apps can run in all super apps, more on this [later](/mini-app-standardization/).
```

## Discovery

Mini apps are often discovered _ad-hoc_ via branded 2D barcodes, which solves an important
offline-to-online challenge, for example, getting from a physical restaurant menu to a payment mini

> app, or from a physical e-scooter to a rental min app. The image below depicts an example of such a
> branded 2d barcode. It is for
> [WeChat's demo mini app](https://github.com/wechat-miniprogram/miniprogram-demo). When the code is
> scanned with the WeChat super app, the mini app launches directly.
> Other super apps will typically not be able to recognize the barcode.

Reviewer: ```suggestion
app, or from a physical e-scooter to a rental min app. The image below shows an example of such a
branded 2D barcode for
[WeChat's demo mini app](https://github.com/wechat-miniprogram/miniprogram-demo). When the code is
scanned with the WeChat super app, the mini app launches directly.
Other super apps will typically not be able to recognize the barcode.
```

<figure class="w-figure">
  <img src="2d-barcode.jpg" alt="WeChat-branded 2d barcode." width="250">
  <figcaption class="w-figure">
    Scanning this 2d barcode with the WeChat app launches a demo mini app.
  </figcaption>
</figure>

Mini apps can also be discovered through regular in-app search in the super app, be shared in chat
messages, or be part of a news item in a news feed. Some super apps have the notion of verified

> accounts that can contain mini apps in their profiles. Mini apps can be highlighted when they are
> (i) physically geographically close, like the mini app of a business in front of which the user
> stands, or (ii) virtually close, like when the user gets directions on a map shown in the super
> app. Frequently used mini apps are available in an app drawer that in many super apps can be
> accessed through a swipe down gesture or through a special section in the super app menu.

Reviewer: ```suggestion
accounts that can contain mini apps in their profiles. Mini apps can be highlighted when they are 
physically geographically close, like the mini app of a business in front of which the user
stands, or virtually close, like when the user gets directions on a map shown in the super
app. Frequently used mini apps are available in an app drawer that in many super apps can be
accessed through a swipe down gesture or through a special section in the super app menu.
```

## The user experience

All super apps have more or less the same user interface for mini apps. A themeable top bar with the
mini app's name, and, in the upper corner of the screen, a close button on the far right preceded by
an action menu that provides access to common features like sharing the app, adding it to a
favorites list or the home screen, reporting abusive apps, providing feedback, and settings. The
screenshot below shows a shopping mini app running in the context of the Alipay super app with the
action menu opened.

<figure class="w-figure">
  <img src="super-app-ui.jpg" alt="The Alipay super app running a shopping mini app with highlighted top bar, action menu button, and close button. The action menu is opened." width="300">
  <figcaption class="w-figure">
    Opened action menu of a shopping mini app running in the Alipay super app.
  </figcaption>
</figure>

## UI paradigms

> Usually there is a bottom tab bar for the mini app's main navigation. Most super app providers offer
> components (see [later](/mini-app-components/)) that help developers implement common UI paradigms quickly,
> like carousels, accordions, progress bars, spinners, switches, maps, etc., which also helps make the
> user experience from mini app to mini app consistent. This is actually encouraged by
> [WeChat's Mini Program Design Guidelines](https://developers.weixin.qq.com/miniprogram/en/design/),
> similar to what Apple incentivizes with its
> [Apple Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/ios/overview/themes/),
> and Google with its [Design for Android](https://developer.android.com/design) recommendations.

Reviewer: ```suggestion
Usually there is a bottom tab bar for the mini app's main navigation. Most [super app providers offer
components](/mini-app-components/) that help developers quickly implement common UI paradigms,
such as carousels, accordions, progress bars, spinners, switches, maps, and so on. This also helps make the
user experience from mini app to super app consistent and it's encouraged by
[WeChat's Mini Program Design Guidelines](https://developers.weixin.qq.com/miniprogram/en/design/),
similar to what Apple incentivizes with its
[Apple Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/ios/overview/themes/),
and Google with its [Design for Android](https://developer.android.com/design) recommendations.
```
